### PR TITLE
Image url fix

### PR DESCRIPTION
**Pull request by daarkdemon**
Fix chapter image url containing whitespaces

Checklist:
- [ ] Updated source's version for individual source changes
- [ ] Updated all sources' versions for template changes
- [ ] Set appropriate `nsfw` value
- [ ] Did not change `id` even if a source's name or language were changed
- [ ] Tested the modifications by running it on the simulator or a test device

### CODE REVIEW

File: src/rust/mangastream/template/src/template.rs [hunk 1]
```diff
@@ -336,7 +336,7 @@ impl MangaStreamSource {
 		} else {
 			for (at, page) in html.select(self.page_selector).array().enumerate() {
 				let page_node = page.as_node();
-				let page_url = page_node.attr(self.page_url).read();
+				let page_url = page_node.attr(self.page_url).read().replace(' ', "%20");
```
Comment (beerpiss) [inline src/rust/mangastream/template/src/template.rs:339]: why not `urlencode` the URL at this point, change the `urlencode` function to not encode `;,/?:@&=+$-_.!~*'()#` and you should be good

**Review by beerpiss — CHANGES_REQUESTED** [review]
You should also bump the version of any sources affected by this issue.

File: src/rust/mangastream/template/src/template.rs (hunk 1 above)
Comment (beerpiss) [reply src/rust/mangastream/template/src/template.rs:339]:
```rust
pub fn urlencode<T: AsRef<[u8]>>(url: T) -> String {
    let bytes = url.as_ref();
    let hex = "0123456789ABCDEF".as_bytes();

    let mut result: Vec<u8> = Vec::with_capacity(bytes.len() * 3);

    for byte in bytes {
        let curr = *byte;
        if curr.is_ascii_alphanumeric() || b";,/?:@&=+$-_.!~*'()#".contains(&curr) {
            result.push(curr);
        } else {
            result.push(b'%');
            result.push(hex[curr as usize >> 4]);
            result.push(hex[curr as usize & 15]);
        }
    }
    String::from_utf8(result).unwrap_or_default()
}
```

Comment (daarkdemon) [reply src/rust/mangastream/template/src/template.rs:339]: Made requested changes

Comment (beerpiss) [reply src/rust/mangastream/template/src/template.rs:339]: you still haven't fixed the specific line I commented on lol

**Review by beerpiss — COMMENTED** [review]
is it only kiryuu that is affected from the bug

File: src/rust/mangastream/template/src/template.rs (hunk 1 above)
Comment (daarkdemon) [reply src/rust/mangastream/template/src/template.rs:339]: I have changed function to url encode for the mentioned line 
Only kiryuu has been affected by this

**Comment by daarkdemon** [comment]
> is it only kiryuu that is affected from the bug

Yes

**Review by beerpiss — APPROVED** [review]

**Pull request merged**